packages/hono/src/build.ts

```diff
@@ -1,51 +1,46 @@
 import { Files, type BuildV3 } from '@vercel/build-utils';
 // @ts-expect-error - FIXME: hono-framework build is not exported
 import { build as nodeBuild } from '@vercel/node';
+import { sep } from 'path';
 
 export const build: BuildV3 = async args => {
   const entrypoint = findEntrypoint(args.files);
 
+  // Introducing new behavior for the node builder where Typescript errors always
+  // fail the build. Previously, this relied on noEmitOnError being true in the tsconfig.json
+  process.env.EXPERIMENTAL_NODE_TYPESCRIPT_ERRORS = '1';
```

Comment:
Does the builder process get reused ever? If so we should switch to just setting this for the this specific `nodeBuild`